src/compiler/compile/Component.ts

```diff
@@ -1551,3 +1553,18 @@ function get_relative_path(from: string, to: string) {
 
 	return from_parts.concat(to_parts).join('/');
 }
+
+function get_basename(filename: string) {
+	return filename.split(/[/\\]/).pop();
+}
+
+function get_sourcemap_source_filename(compile_options: CompileOptions) {
+	if (compile_options.filename) {
```

Comment:
nit: I might reverse the direction here to avoid having to indent everything:
```
if (!compile_options.filename) return;
```

Reply:
went with return null to keep the spirit of the previous implementation, but i'm indifferent. if early return is preferred style in svelte, i'm in

Reply:
You could still return `null`. I don't have an opinion about that

Reply:
done, it's more clear now. thanks